ext/rotoscope/rotoscope.c

```diff
@@ -119,44 +119,35 @@ static rs_class_desc_t class2str(VALUE klass)
   return real_class;
 }
 
-static char *caller_location(int start)
+static rs_callsite_t caller_location(int start)
 {
   VALUE argv[2] = {INT2NUM(start), INT2NUM(1)};
-  VALUE res = rb_funcallv(rb_mKernel, id_caller_locations, 2, argv);
-  return RSTRING_PTR(rb_inspect(rb_ary_entry(res, 0)));
+  VALUE locations = rb_funcallv(rb_mKernel, id_caller_locations, 2, argv);
+  VALUE location = rb_ary_entry(locations, 0);
+  VALUE path = rb_funcall(location, id_path, 0);
```

Comment:
👍 Didn't realize `Thread::Backtrace::Location` exposed the info like this. I think I whipped up the original code as a PoC and never revisited. Good idea.